code/game/turfs/simulated/floor/light_floor.dm

```diff
@@ -111,3 +111,44 @@
 	var/color_save = color
 	..()
 	color = color_save
+
+// These tiles change color every now and then
+/turf/simulated/floor/light/disco
+	floor_tile = /obj/item/stack/tile/disco_light
+	can_modify_colour = FALSE	// This tile has a set of colours, thus no interfering
+	var/list/available_colors = list("#d41e3c", "#ed7b39", "#fff540", "#77b02a", "#488bd4", "#b0fff1", "#94007a", "#ff417d")	// These are all the colors we can ever become
+	var/current_color	// This is our current color, don't pick it again (integer for faster list accessing)
+
+// We pick a random color when we are spawned
+/turf/simulated/floor/light/disco/Initialize()
+	START_PROCESSING(SSobj, src)
+	current_color = rand(1, length(available_colors))
+	..()
+
+// The animation happens in this proc
+/turf/simulated/floor/light/disco/proc/change_color()
+	animate_fade_to_color_fill(src, pick_color(), 2)
```

Comment:
Not sure on how I feel about `animate()` for all of these tiles every 2 seconds. Will have to see how it handles at runtime I guess.

Reply:
This is the reason why I made them to be limited - players cannot craft it, only 30 exists (can be changed) and it is only obtainable via traders. However, if it is too much, gonna tweak it of course

Reply:
> This is the reason why I made them to be limited - players cannot craft it, only 30 exists (can be changed) and it is only obtainable via traders. However, if it is too much, gonna tweak it of course

That should be just fine